protocols/kad/src/kbucket.rs

```diff
@@ -525,6 +540,30 @@ mod tests {
         }
     }
 
+    #[test]
+    fn bucket_contains_range() {
```

Comment:
Another property to check may be that the ranges of all buckets partition the keyspace, i.e. are non-overlapping and exhaustive. I leave it up to you whether you want to include such a test.

Reply:
Good idea. Added in https://github.com/libp2p/rust-libp2p/pull/1680/commits/501e80c5c73adbcceff4e01848874a560b560ee3 plus https://github.com/libp2p/rust-libp2p/pull/1680/commits/f8ba555ae4a75fd4d9f263ff86c882b66dd43b25 fixing the overflow panic it revealed. Thanks!